### PR TITLE
Endpoint delete an item

### PR DESCRIPTION
**Pull request by wally-yawn**
There is a lot here one I got to cascading delete that had to be updated...

### CODE REVIEW

**Review by Jonathan-Atkins — APPROVED** [review]

File: app/controllers/api/v1/customers_controller.rb
Comment (James-Cochran) [inline app/controllers/api/v1/customers_controller.rb:1]: can we refactor the invoices iteration using an AR joins?
Also, I believe should be moved to the model.

File: app/models/customer.rb
Comment (James-Cochran) [inline app/models/customer.rb:1]: I know you mentioned in morning stand up, but I don't remember.  Why did you remove the customer belongs_to merchant association?

File: app/models/merchant.rb
Comment (James-Cochran) [inline app/models/merchant.rb:1]: I think has_many items singularly along with the dependent destroy is redundant.

**Review by James-Cochran — APPROVED** [review]
Approved, but some design or refactor questions

File: app/controllers/api/v1/customers_controller.rb
Comment (wally-yawn) [reply app/controllers/api/v1/customers_controller.rb:1]: yup, I've added cards to the backlog for those refactors

File: app/models/customer.rb
Comment (wally-yawn) [reply app/models/customer.rb:1]: it broke a lot of the delete logic because it added a foreign key and we could get there without it

File: app/models/merchant.rb
Comment (wally-yawn) [reply app/models/merchant.rb:1]: hmmm, that's not there on my branch, I think that's a funky merge thing. If it's still there when it's merged from main I'll push up a new PR to fix that.
